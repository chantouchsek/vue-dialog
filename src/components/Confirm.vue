<template>
  <div class="confirm-box">
    <div v-if="!!title" class="confirm-title">
      <span class="title">
        {{ title }}
      </span>
    </div>
    <div class="confirm-message" v-html="text" />
    <div class="confirm-buttons">
      <div class="column">
        <dialog-actions :actions="actions" />
      </div>
    </div>
  </div>
</template>

<script>
import Confirmable from '../mixins/confirmable'
import DialogActions from './DialogActions.vue'

export default {
  components: { DialogActions },
  layout: ['default', { width: 450 }],
  mixins: [Confirmable]
}
</script>

<style>
  .confirm-box {
    background: #ffffff;
  }

  .confirm-box .confirm-title {
    padding: 14px
  }

  .confirm-box .confirm-title .title {
    font-size: 24px;
  }

  .confirm-box .confirm-message {
    padding: 0 20px;
  }

  .confirm-box .confirm-buttons {
    text-align:  right;
    padding: 10px 20px;
  }

  /* .confirm-box .confirm-buttons .button {
    font-size: 20px;
    padding-left: 20px;
  } */
</style>
