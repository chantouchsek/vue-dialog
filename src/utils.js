export function destroyVueElement (vm) {
  if (vm && !vm._isDestroyed && (typeof vm.$destroy === 'function')) {
    vm.$destroy()
  }
}

export function findContainer (container) {
  let found
  if (typeof container === 'string') {
    found = document.querySelector(container)
  } else {
    found = container
  }
  if (!found) {
    found = document.body
  }
  return found
}

// todo
// export function middlewareSeries (promises, appContext) {
//   if (!promises.length || appContext._redirected || appContext._errored) {
//     return Promise.resolve()
//   }
//   return promisify(promises[0], appContext)
//     .then(() => {
//       return middlewareSeries(promises.slice(1), appContext)
//     })
// }
