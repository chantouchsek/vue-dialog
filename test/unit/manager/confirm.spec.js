import Dialog from '../../../src/dialog'
import DialogManager from '../../../src/manager'
import Confirm from '../../../src/components/Confirm'
import { wrap } from '../../utils'
import Returnable from '../../../src/mixins/returnable'
import Vue from 'vue'

describe('manager', () => {
  let manager

  beforeAll(() => {
    manager = new DialogManager({ context: { store: {} } })
    manager.component('confirm', Confirm)
  })

  test('Create manager instance', () => {
    expect(manager).toBeInstanceOf(DialogManager)
  })

  test('returnable', async () => {
    const dlg = new Dialog({
      mixins: [Returnable],
      template: '<p></p>'
    })
    await dlg.show()
    expect(typeof dlg.vmd.return).toBe('function')
    expect(dlg.element).toMatchSnapshot()
    setTimeout(() => {
      dlg.vmd.return(123)
    }, 5)
    const res = await dlg.wait()
    expect(res).toBe(123)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Check confirm', async () => {
    const dlg = await manager.confirm({
      text: 'test',
      actions: ['ok', 'cancel']
    })
    const wrapper = wrap(dlg.vm)
    expect(dlg.vm.$el).toMatchSnapshot()
    setTimeout(() => {
      wrapper.find('[action-key=ok]').trigger('click')
    }, 5)
    const res = await dlg.wait()
    expect(res).toBe('ok')
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Check confirm with btns true|false', async () => {
    const dlg = await manager.confirm({
      text: 'test',
      actions: { true: 'Yes', false: 'No' }
    })
    const wrapper = wrap(dlg.vm)
    console.warn('wrapper', wrapper)
    expect(dlg.element).toMatchSnapshot()
    expect(wrapper.find('[action-key=true]').exists()).toBe(true)
    expect(wrapper.find('[action-key=false]').exists()).toBe(true)
    setTimeout(() => {
      wrapper.find('[action-key=true]').trigger('click')
    }, 10)
    const res = await dlg.wait()
    expect(res).toBe(true)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Check confirm with handler functions', async () => {
    const dlg = await manager.confirm({
      text: 'test',
      actions: {
        false: 'No',
        true: {
          text: 'Yes',
          handler: () => {
            return new Promise((resolve) => {
              setTimeout(() => resolve({ msg: 'foo' }), 10)
            })
          }
        }
      }
    })
    const wrapper = wrap(dlg.vm)
    expect(dlg.element).toMatchSnapshot()
    setTimeout(() => {
      wrapper.find('[action-key=true]').trigger('click')
      Vue.nextTick(() => {
        expect(wrapper.find('[action-key=true]').element.hasAttribute('disabled')).toBe(true)
        expect(dlg.element).toMatchSnapshot()
      })
    }, 5)
    const res = await dlg.wait()
    expect(res).toEqual({ msg: 'foo' })
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })
})
