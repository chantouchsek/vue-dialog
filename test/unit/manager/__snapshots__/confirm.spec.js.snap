// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Check confirm 1`] = `
<div
  class="dialog-layout"
>
  <div
    class="confirm-box"
  >
    <!---->
     
    <div
      class="confirm-message"
    >
      test
    </div>
     
    <div
      class="confirm-buttons"
    >
      <div
        class="column"
      >
        <span>
          <button
            action-key="ok"
            class="button"
          >
            
  ok

          </button>
          <button
            action-key="cancel"
            class="button"
          >
            
  cancel

          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`manager Check confirm with btns true|false 1`] = `
<div
  class="dialog-layout"
>
  <div
    class="confirm-box"
  >
    <!---->
     
    <div
      class="confirm-message"
    >
      test
    </div>
     
    <div
      class="confirm-buttons"
    >
      <div
        class="column"
      >
        <span>
          <button
            action-key="true"
            class="button"
          >
            
  Yes

          </button>
          <button
            action-key="false"
            class="button"
          >
            
  No

          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`manager Check confirm with handler functions 1`] = `
<div
  class="dialog-layout"
>
  <div
    class="confirm-box"
  >
    <!---->
     
    <div
      class="confirm-message"
    >
      test
    </div>
     
    <div
      class="confirm-buttons"
    >
      <div
        class="column"
      >
        <span>
          <button
            action-key="false"
            class="button"
          >
            
  No

          </button>
          <button
            action-key="true"
            class="button"
          >
            
  Yes

          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`manager Check confirm with handler functions 2`] = `
<div
  class="dialog-layout"
>
  <div
    class="confirm-box"
  >
    <!---->
     
    <div
      class="confirm-message"
    >
      test
    </div>
     
    <div
      class="confirm-buttons"
    >
      <div
        class="column"
      >
        <span>
          <button
            action-key="false"
            class="button"
            disabled="disabled"
          >
            
  No

          </button>
          <button
            action-key="true"
            class="button loading"
            disabled="disabled"
          >
            
  Yes

          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`manager returnable 1`] = `
<div
  class="dialog-layout"
>
  <p />
</div>
`;
